### ISSUE TITLE
association's getters fail for new record

### ISSUE THREAD

**Issue by Verdier** [sequelize/sequelize#3497]
Using an association's getter on a new record, for exemple `project.getPictures()` result in querying the target with :
```javascript
options.where = new Utils.and([
      new Utils.where(
        this.target.rawAttributes[this.association.identifier],
        this.instance[this.source.primaryKeyAttribute]
      ),
      scopeWhere,
      options.where
    ]);
```
where `this.instance[this.source.primaryKeyAttribute]` is `null`. That will finally crash in `abstract/query-generator.js:1495:18` with `Cannot read property '_isSequelizeMethod' of null`.

I don't know what is the best fix for that, probably the getter should return an empty list if `instance.isNewRecord`?

**Comment by mickhansen** [comment]
New record as in newly created or new record as one that has only been `.build()`?

**Comment by Verdier** [comment]
My issue is for one that has only been `.build()`. There is other type of record for whose `isNewRecord` returns true?

**Comment by mickhansen** [comment]
No it should only be for build :)
A saner error when there's no primary key value available would be nice. However an association getter would still work if you built it with a primary key value.

**Comment by Verdier** [comment]
The primary key is often auto generated... you couldn't set it when you built it. Do you want a pull request for a saner error ?

**Comment by mickhansen** [comment]
@Verdier Mostly, but some people cheat a bit by using `Model.build({id: value})` in API's to skip doing a find :)

A PR is always appreciated.

**Comment by Verdier** [comment]
Fixed by a42b38b7d4214a091232d6e437ceda6803563dcd.
